datacube/api/query.py

```diff
@@ -264,15 +268,34 @@ def _to_datetime(t):
 
     return pandas_to_datetime(t, utc=True, infer_datetime_format=True).to_pydatetime()
 
-
 def _time_to_search_dims(time_range):
-    if hasattr(time_range, '__iter__') and len(time_range) == 2:
-        time_range = Range(_to_datetime(time_range[0]), _to_datetime(time_range[1]))
-        if time_range[0] == time_range[1]:
-            return time_range[0]
-        return time_range
-    else:
-        return _to_datetime(time_range)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", UserWarning)
+        if hasattr(time_range, '__iter__') and len(time_range) == 2:
```

Comment:
There's a lot more complexity in this function than I think is necessary, It's quite hard to read and understand, which is something we want to avoid.

Can we sit down and go through some ways to simplify it before we merge this in.